examples/06-suspense-like/src/app.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from "react";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import { WithStyles, withStyles } from "@material-ui/core/styles";
+import { trackPromise, usePromiseTracker } from "react-promise-tracker";
+import { Quote, getQuote, getPicture } from "./api";
+import { styles } from "./styles";
+
+
+const Suspense: React.FC<{ tag: string, className?: string }> = ({ tag, className, children }) => {
+  const { promiseInProgress } = usePromiseTracker({ area: tag, delay: 0 });
+  return promiseInProgress ? <CircularProgress className={className} size={60}/> : <>{children}</>;
+};
+
+const AppInner: React.FC<WithStyles<typeof styles>> = ({classes}) => {
+  const [quote, setQuote] = React.useState<Quote>({ body: "", author: "" });
+  const [picture, setPicture] = React.useState();
+  const loadData = React.useCallback(() => {
```

Comment:
Why did you use `useCallback`?

`useCallback` is to memoize the function by some params, but if you pass `[]` only memoize once.

Then, you are using it in `useEffect` but with `[]` as second parameter, that is, we are only calling `loadData` on componentDidMount.

Am I missing something?

Thank you.

Reply:
Cause I want to intentionally keep the same callback reference between re-renders. `loadData` does also serve as a handler for the button `onClick`.